pkg/kube/kube_handler.go

```diff
@@ -881,56 +861,57 @@ func ValidateOCIAppRepository(appRepo *v1alpha1.AppRepository, cli httpclient.Cl
 	return true, nil
 }
 
-func (r HelmNonOCIValidator) IsValid(cli httpclient.Client) (*ValidationResponse, error) {
+// HttpValidator is an interface for checking the validity of an AppRepo via Http requests.
+type HttpValidator interface {
+	// Validate returns a validation response.
+	Validate(cli httpclient.Client) (*ValidationResponse, error)
+}
+
+// HelmNonOCIValidator is an HttpValidator for non-OCI Helm repositories.
+type HelmNonOCIValidator struct {
+	Req *http.Request
+}
+
+func (r HelmNonOCIValidator) Validate(cli httpclient.Client) (*ValidationResponse, error) {
 
 	response := &ValidationResponse{}
 
-	res, err := cli.Do(r.req)
+	res, err := cli.Do(r.Req)
 	if err != nil {
 		// If the request fail, it's not an internal error
 		return &ValidationResponse{Code: 400, Message: err.Error()}, nil
 	}
-	body, err := ioutil.ReadAll(res.Body)
-	if err != nil {
-		return nil, fmt.Errorf("Unable to parse validation response. Got: %w", err)
-	}
-	response = &ValidationResponse{Code: res.StatusCode, Message: string(body)}
+	response = &ValidationResponse{Code: res.StatusCode, Message: "OK"}
 	if response.Code != 200 {
-		return response, nil
+		body, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			return nil, fmt.Errorf("Unable to parse validation response. Got: %w", err)
+		}
+		response.Message = string(body)
 	}
 
 	return response, nil
 }
 
-func (r HelmNonOCIValidator) getReq() *http.Request {
-	return r.req
+type HelmOCIValidator struct {
+	AppRepo *v1alpha1.AppRepository
 }
 
-func (r HelmNonOCIValidator) getAppRepo() *v1alpha1.AppRepository {
-	return nil
-}
-func (r HelmOCIValidator) IsValid(cli httpclient.Client) (*ValidationResponse, error) {
+func (r HelmOCIValidator) Validate(cli httpclient.Client) (*ValidationResponse, error) {
```

Comment:
Since the interface method has a signature that isn't returning a bool, but rather returning a validation response, I've renamed the method (as `IsValid` implies a single bool return).